### PR TITLE
kcp-front-proxy: implement options pattern

### PR DESCRIPTION
**Pull request by sttts**
Reuse `SecureServing` from kube, and align default options with `kcp start`.

### CODE REVIEW

File: cmd/kcp-front-proxy/main.go [hunk 1]
```diff
@@ -40,26 +66,39 @@ forwards Common Name and Organizations to backend API servers in HTTP headers.
...
 	}
 
-	// setup klog
-	fs := goflags.NewFlagSet("klog", goflags.PanicOnError)
-	klog.InitFlags(fs)
-	cmd.PersistentFlags().AddGoFlagSet(fs)
+	options.AddFlags(cmd.Flags())
 
-	cmd.PersistentFlags().StringVar(&server.ListenAddress, "listen-address", ":8083", "Address and port for the proxy to listen on")
-	cmd.PersistentFlags().StringVar(&server.ClientCACert, "client-ca-cert", "certs/kcp-client-ca-cert.pem", "CA cert used to validate client certs")
```
Comment (csams) [inline cmd/kcp-front-proxy/main.go:53]: @sttts The `--client-ca-file` option isn't available to verify client certs.

**Review by csams — CHANGES_REQUESTED** [review]
Thanks for the changes and opportunity to see a small part of how secure serving is set up!

How do I configure the CA to validate client certs? The `--client-ca-file` option isn't available.

File: cmd/kcp-front-proxy/main.go (hunk 1 above)
Comment (sttts) [reply cmd/kcp-front-proxy/main.go:53]: that's via authentication. Have not wired that, but left your code in place. Might be worth to explore using the kube authenticators too.

Comment (sttts) [reply cmd/kcp-front-proxy/main.go:53]: Would be good to know whether I broke the old code or whether it still works. We can attempt to use kube's client cert auth in follow-up.

Comment (csams) [reply cmd/kcp-front-proxy/main.go:53]: The old code was validating client certs, so it's broken. :) I chatted with @ncdc and got some pointers for wiring up kube's client cert auth, though. I'll merge and start on a follow up.

**Review by csams — APPROVED** [review]
Will follow up with a PR to use kube's client cert auth.

**Pull request merged**